components/FooterApp.tsx

```diff
@@ -1,10 +1,58 @@
-import Link from "next/link";
-import { Github, Linkedin } from "lucide-react";
-import XIcon from "./ui/XIcon";
+'use client'; // Add this directive at the top
+
+import Link from 'next/link';
+import { Github, Linkedin } from 'lucide-react';
+import XIcon from './ui/XIcon';
+import { useState } from 'react';
+import emailjs from 'emailjs-com';
 
 function FooterApp() {
+  const [email, setEmail] = useState('');
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState('');
+  const [success, setSuccess] = useState('');
+
+  const handleSubscribe = async (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault(); // Prevent the default form submission
+
+    if (!email) {
+      alert('Please enter a valid email address.');
+      return;
+    }
+
+    console.log("Email sent to", email);
+
+    const emailData = {
+      service_id:"your_service_id",
+      template_id: "your_template_id",
+      user_id: "your_user_id",
+      template_params: { user_email: email }
+    };
+
+    setLoading(true);
+    setError('');
+    setSuccess('');
+
+    try {
+      const response = await emailjs.send(
```

Comment:
instead of sending them email store their email in mongodb  and then with better template send them email on saving email success you have subscribed to our newsletter

Reply:
ok